lib/datadog/tracing/utils.rb

```diff
@@ -43,6 +43,34 @@ def self.reset!
       end
 
       private_class_method :id_rng, :reset!
+
+      # The module handles bitwise operation for trace id
+      module TraceId
+        MAX = (1 << 128) - 1
+
+        module_function
+
+        def next_id
+          return Utils.next_id unless Datadog.configuration.tracing.trace_id_128_bit_generation_enabled
+
+          concatenate(
+            Time.now.to_i << 32,
```

Comment:
We have a time provider, that allows users to mock the `ddtrace` time provider for testing, we should use that:
```suggestion
            Core::Utils::Time.now.to_i << 32,
```